apps/notion-api/src/app/api/authentication/getAccessToken/route.ts

```diff
@@ -0,0 +1,80 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import { NextResponse, NextRequest } from 'next/server';
+import { PrismaClient } from '@prisma/client';
+import { getNotionInfo } from '@/lib/auth';
+
+const prisma = new PrismaClient();
+
+type RequestBody = {
+    accessCode : string;
+    organizationId : string;
+}
+
+type AccessTokenResponse = {
+    workspace_id : string;
+    workspace_name : string; 
+    access_token : string;
+}
+
+export async function POST (req: NextRequest) {
+        
+    const body : RequestBody = await req.json();
+    const { accessCode, organizationId } = body;
+
+    const encoded = Buffer.from(`${getNotionInfo().notionClientID}:${getNotionInfo().notionClientSecret}`).toString("base64");
+
+    const response = await fetch("https://api.notion.com/v1/oauth/token", {
+        method: "POST",
+        headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Basic ${encoded}`,
+    },
+        body: JSON.stringify({
+        grant_type: "authorization_code",
+        code: accessCode,
+        redirect_uri: getNotionInfo().notionRedirectUrl
+        }),
+    });
+
+    const data : AccessTokenResponse = await response.json();
+
+    if (response.status === 200) {
```

Comment:
Prefer [early return](https://gomakethings.com/the-early-return-pattern-in-javascript/) here 
```suggestion
    if (!response.ok === 200) {
      throw new Error(`Could not retrieve access token from notion for organizationId=${organizationId}`)
    }
```